botorch/acquisition/hentropy_search.py

```diff
@@ -0,0 +1,282 @@
+#!/usr/bin/env python3
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+r"""
+This file has an implementation of Batch H-Entropy Search (qHES) via
+one-shot optimization as introduced in [Neiswanger2022]_.
+
+The authors adopt a generalized definition of entropy from past work
+in Bayesian decision theory, which proposes a family of decision-theoretic
+entropies parameterized by a problem-specific loss function and
+action set. Therefore, the method allow the development of a common
+acquisition optimization procedure, which applies generically
+to many members of this family (where each member is induced by a
+specific loss function and action set).
+
+.. [Neiswanger2022]
+    W. Neiswanger, L. Yu, S. Zhao, C. Meng, S. Ermon. Generalizing Bayesian
+    Optimization with Decision-theoretic Entropies. Appears in Proceedings
+    of the 36th Conference on Neural Information Processing Systems
+    (NeurIPS 2022)
+"""
+
+from __future__ import annotations
+
+from typing import Any, Dict, Optional
+
+import torch
+import torch.nn as nn
+from botorch.acquisition.acquisition import OneShotAcquisitionFunction
+from botorch.acquisition.monte_carlo import MCAcquisitionFunction
+from botorch.models.model import Model
+from botorch.sampling.base import MCSampler
+from botorch.sampling.normal import SobolQMCNormalSampler
+from botorch.utils.transforms import t_batch_mode_transform
+from torch import Tensor
+
+
+class qHEntropySearch(MCAcquisitionFunction, OneShotAcquisitionFunction):
+    r"""H-Entropy Search using one-shot optimization."""
```

Comment:
Nit: This doesn't actually need to be a raw string, and some linters will complain if a raw string is used unnecessarily. It's fine for BoTorch though.